File: java/iroha_android/src/main/java/org/hyperledger/iroha/android/offline/OfflineProofRequestResult.java
package org.hyperledger.iroha.android.offline;

import java.util.Objects;

/** Canonical proof payload returned by Torii (`/v1/offline/transfers/proof`). */
public final class OfflineProofRequestResult {

  private final OfflineProofRequestKind kind;
  private final String json;

  public OfflineProofRequestResult(
      final OfflineProofRequestKind kind, final String jsonPayload) {
    this.kind = Objects.requireNonNull(kind, "kind");
    this.json = Objects.requireNonNull(jsonPayload, "jsonPayload");
  }

  public OfflineProofRequestKind kind() {
    return kind;
  }

  /**
   * Canonical JSON representation of the proof payload; this string can be passed directly to the
   * FASTPQ prover.
   */
  public String json() {
    return json;
  }
}

File: kotlin/core-jvm/src/main/java/org/hyperledger/iroha/sdk/offline/OfflineProofRequestResult.kt
package org.hyperledger.iroha.sdk.offline

/** Canonical legacy proof payload retained for local fixture parsing. */
class OfflineProofRequestResult(
    val kind: OfflineProofRequestKind,
    /**
     * Canonical JSON representation of the proof payload; this string can be passed directly to the
     * FASTPQ prover.
     */
    val json: String,
)
